include/git2/common.h

```diff
@@ -59,6 +59,12 @@ typedef size_t size_t;
 #define GIT_WIN32 1
 #endif
 
+#if defined(__has_feature)
+#if __has_feature(address_sanitizer)
+#define GIT_ASAN 1
```

Comment:
I'm not sure I understand what's going on here.  When is `__has_feature(address_sanitizer)` true?  It seems like it's true when `address_sanitizer` is available, not when libgit2 was built with it, no?

If your goal is to provide some functionality to callers about whether libgit2 was built with address_sanitizer, then please have cmake set `-DGIT_ASAN` and expose a function that will check and return that.  (Probably a query key on `git_libgit2_opts` makes sense.)